src/test/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannelTest.java

```diff
@@ -46,32 +47,35 @@
 
 @RunWith(Parameterized.class)
 public class TopicPartitionChannelTest {
```

Comment:
Ok I kind of tunnel visioned and made a lot more changes in here than necessary - mostly enforcing more mocks and all of that. Happy to roll this back if needed, but will take time. I do think the additional verification is good to have though